### PR TITLE
General cleanup of the charm

### PR DESCRIPTION
**Pull request by jakobilobi**
While preparing to work with the COS Lite integration I was reading a lot of stuff in the charm and took a cleanup pass while at it. Gathered it all in one PR to only produce a single revision update:

- Updated handling of Juju status
- Updated function arguments for more clarity regarding types
- Cleaned some imports/strings/other small stuff
- Updated `ops` to 2.4.1 (which did solve an error i got along the way)

All changes except the `ops` upgrade should be 100% harmless but since 2.0 -> 2.4.1 is quite a big step in version we might want to do some validation/testing before any merge. What do you guys think is reasonable here?

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -12,15 +12,13 @@
...
+from requests.exceptions import ConnectionError as RequestsConnectionError
 from urllib3.exceptions import NewConnectionError, MaxRetryError
 import time
 import re
 
-from ops.charm import CharmBase
-from ops.main import main
-from ops.framework import StoredState
+import ops
+from ops.charm import CharmBase, ActionEvent
```
Comment (Maharacha) [inline src/charm.py:21]: Isn't this importing same stuff twice?

**Comment by Maharacha** [comment]
It's a nice cleanup! I think it is enough with test deploy it, and see that the service starts up fine.

**Review by Maharacha — APPROVED** [review]

**Comment by jakobilobi** [comment]
> It's a nice cleanup! I think it is enough with test deploy it, and see that the service starts up fine.

Already done then, this is what I've been using when working with COS Lite :+1:

File: src/charm.py (hunk 1 above)
Comment (jakobilobi) [reply src/charm.py:21]: Well kind of, I'll clean it up before merging

**Pull request merged**